syntax = "proto3";

package com.nus.dbsytem.falcon.v0;
option go_package = "/v0";

//import public "vo.proto";

// This message denotes the pearson feature selection parameters
message PearsonFeatSelParams {
    // selected samples file
    string selected_samples_file = 1;
    // selected predictions file
    string selected_predictions_file = 2;
    // the sample weights file
    string sample_weights_file = 3;
    // number of samples generated or selected
    int32 num_samples = 4;
    // number of classes in classification, set to 1 if regression
    int32 class_num = 5;
    // the label id to be explained
    int32 class_id = 6;
    // feature selection method,
    string feature_selection = 7;
    // number of features to be explained in the interpret model
    int32 num_explained_features = 8;
    // selected features to be saved
    string selected_features_file = 9;
}
